src/main/java/org/eclipsefoundation/adopters/service/impl/DefaultAdopterService.java

```diff
@@ -0,0 +1,213 @@
+
+import com.google.gson.Gson;
+
+import io.quarkus.runtime.Startup;
+
+/**
+ * Retrieves the adopters information from the filesystem on service start, and
+ * provides copies to requesting callers.
+ * 
+ * @author Martin Lowe
+ *
+ */
+@Startup
+@ApplicationScoped
+public class DefaultAdopterService implements AdopterService {
+	private static final Logger LOGGER = LoggerFactory.getLogger(DefaultAdopterService.class);
+
+	@ConfigProperty(name = "eclipse.adopters.dirpath", defaultValue = "/tmp/quarkus")
+	String adoptersLocation;
+	@ConfigProperty(name = "eclipse.adopters.filename", defaultValue = "adopters.json")
+	String adoptersName;
+	@ConfigProperty(name = "eclipse.adopters.retry.count", defaultValue = "3")
+	int retryCount;
+	@ConfigProperty(name = "eclipse.adopters.retry.timeoutInMS", defaultValue = "1000")
+	long retryTimeout;
+
+	private List<Adopter> adopters;
+	private WatchService watcher;
+
+	private ReentrantLock lock;
+	private boolean running;
+
+	@PostConstruct
+	public void init() throws IOException {
+		this.lock = new ReentrantLock();
+		this.adopters = new ArrayList<>();
+		Path adoptersDir = Paths.get(adoptersLocation);
+		Path adoptersFile = Paths.get(adoptersLocation, adoptersName);
+		if (Files.exists(adoptersDir)) {
+			// read in the initial file if it exists
+			if (Files.exists(adoptersFile)) {
+				LOGGER.debug("Found an adopters file at path {}, reading in", adoptersFile);
+				readInFile(adoptersFile);
+			}
+
+			watcher = FileSystems.getDefault().newWatchService();
+			// register the adopter location to watch
+			LOGGER.debug("Registering file watcher on directory {}", adoptersDir);
+			adoptersDir.register(watcher, ENTRY_CREATE, ENTRY_DELETE, ENTRY_MODIFY);
+
+			// start file watching on a new thread
+			new Thread(this::update).start();
+		} else {
+			throw new IllegalArgumentException();
+		}
+	}
+
+	@PreDestroy
+	public void shutdown() {
+		this.running = false;
+	}
+
+	/**
+	 * Update loop, will loop until shutdown is triggered or the service is stopped.
+	 * 
+	 * @throws IOException
+	 */
+	private void update() {
+		// soft lock this to a single thread
+		if (!this.lock.isLocked()) {
+			this.lock.lock();
+			try {
+				// set the state to running at the start
+				this.running = true;
+				// watch for running state to change for gentle shutdowns
+				while (this.running) {
+					WatchKey key = null;
+					try {
+						key = watcher.take();
+						// if there was an error while watching for changes, try again
+						if (key == null) {
+							continue;
+						}
+
+						// pass a filtered list of events to be processed
+						@SuppressWarnings("unchecked")
+						List<WatchEvent<Path>> events = key.pollEvents().stream()
+								.filter(e -> e.kind() != OVERFLOW && e.kind().type() == Path.class)
+								.map(e -> (WatchEvent<Path>) e)
+								.filter(e -> ((Path) e.context()).getFileName().toString().equals(adoptersName))
+								.collect(Collectors.toList());
+						processEvents(events);
```

Comment:
```suggestion
								.forEach(this::processEvent);
```

and change `processEvents(List<WatchEvent<Path>>)` to `processEvent(WatchEvent<Path>)`, ie that only deal with 1 event